cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CaReconciler.java

```diff
@@ -373,6 +382,47 @@ Future<Void> rollingUpdateForNewCaKey() {
         }
     }
 
+    /**
+     * Verify that all the pods are already trusting the new CA certificate signed by a new CA key.
+     * It checks each pod's CA key generation, compared with the new CA key generation.
+     * When the trusting phase is not completed (i.e. because CO stopped), it needs to be recovered from where it was left.
+     */
+    Future<Void> isClusterCaFullyTrusted() {
+        // Building the selector for Kafka related components
+        Labels labels =  Labels.forStrimziCluster(reconciliation.name()).withStrimziKind(Kafka.RESOURCE_KIND);
+
+        isClusterCaFullyTrusted = true;
+        return podOperator.listAsync(reconciliation.namespace(), labels)
```

Comment:
Just a thought ... the list is a relatively expensive operation ... and we will now do it twice for every reconciliation just in the `CaReconciler`. Would it make sense to list the pods once at the beginning to establish both:
* Whether the new CA is fully trusted
* Whether the the old CA can be removed

And then in two separate steps handle the flags?

Reply:
That's a good point. My only thought is the following ...
Between the trusting check and the CA cert removal there is the rolling for new CA key. While we don't expect changes in the `Pod` definition, they will be "technically" different pods, even with different names (actually ZK and K pods will have same names but EO, CC, KE will not). We should also discard the pods in "terminating" state when listing because they won't be there anymore during the CA cert removal.

Reply:
So, what about this comment?

Reply:
Actually I was waiting for your feedback on my concern. The list of pods at the beginning could be different from the one across the steps (at least the first 2 which are done in the same reconciliation).

Reply:
But is it really possible that that changes the result?

Reply:
maybe not. the two steps you mentioned above are checking a couple annotation that won't change across the reconcile. What bothers me is just that they will really be different pods between step 1 and 2 :-)
but if you like I can make it listing one time and storing in the CaReconciler as a member field.

Reply:
I have pushed changes with listing the pods just one time at the beginning of the CA reconciliation.